server/revision/revision.controller.js

```diff
@@ -2,20 +2,30 @@
 
 const { Revision, User } = require('../shared/database');
 const { resolveStatics } = require('../shared/storage/helpers');
+const revisionService = require('./revision.service');
 
 function index({ repository, query }, res) {
-  const { limit, offset, entity, entityId } = query;
-  const where = { repositoryId: repository.id };
-  if (entity) {
-    where.entity = entity;
-    where.state = { id: entityId };
-  }
+  const { limit, offset, entity, publishedOn } = query;
```

Comment:
Limit and offset should be taken from opts (i.e. see repository ctrl index; processed within me). Consider making a publishedOn param a bit more generic / less related to this feature (i.e. before / endDate ?)

Reply:
There's a lot of querying logic specific to this feature. If the param was more generic, do you think it would be weird that the service method was picked based on it? 😅 
It seemed to me that publish-related param plays well (somewhat generic) with the business logic we have, while not leaking the query logic to the API consumer. 🙃